### PR TITLE
SplitButton Minor Fixes

### PR DESCRIPTION
**Pull request by Boris-Em**
#### Pull request checklist

This PR has two intended goals for the SplitButton component:
- Fixes minor styling issues
- Add a new `splitButton` boolean property to ensure that we want to render the button as a SplitButton.

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx [hunk 1]
```diff
@@ -30,7 +30,7 @@ export interface IBaseButtonState {
 export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState> implements IButton {
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx]: dont forget change file.

File: packages/office-ui-fabric-react/src/components/Button/Button.Props.ts [hunk 2]
```diff
@@ -76,11 +76,16 @@ export interface IButtonProps extends React.HTMLAttributes<HTMLButtonElement | H
...
    * Props for button menu. Providing this will default to showing the menu icon. See menuIconProps for overriding
-   * how the default icon looks. Providing this in addition of onClick will render a SplitButton.
+   * how the default icon looks. Providing this in addition of onClick and setting splitButton to true will render a SplitButton.
    */
   menuProps?: IContextualMenuProps;
 
   /**
+   * If set to true, and if menuProps and onClick are provided, the button will render as a SplitButton. Defaults to false.
+   */
+  splitButton?: boolean;
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Button/Button.Props.ts]: Maybe we could just shorten this to `split` so that the jsx would look a bit less redundant, like this:

```tsx
<PrimaryButton split text='foo' />
```

Comment (Boris-Em) [inline packages/office-ui-fabric-react/src/components/Button/Button.Props.ts]: I like that, thanks!

File: packages/office-ui-fabric-react/src/components/Button/Button.Props.ts [hunk 3]
```diff
@@ -76,11 +76,16 @@ export interface IButtonProps extends React.HTMLAttributes<HTMLButtonElement | H
...
    * Props for button menu. Providing this will default to showing the menu icon. See menuIconProps for overriding
-   * how the default icon looks. Providing this in addition of onClick will render a SplitButton.
+   * how the default icon looks. Providing this in addition of onClick and setting the split property to true will render a SplitButton.
    */
   menuProps?: IContextualMenuProps;
 
   /**
+   * If set to true, and if menuProps and onClick are provided, the button will render as a SplitButton. Defaults to false.
+   */
+  split?: boolean;
```
Comment (jair-cazarin) [inline packages/office-ui-fabric-react/src/components/Button/Button.Props.ts]: shall we warn if the user provides this but not the others? Ideally we would not have them as optional...like using union types.

Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/Button/Button.Props.ts]: if you provided an onClick and menuProps, how would it not be a split button? Better question is what happens if you supply onClick and menuProps but have split as false?

Comment (Boris-Em) [inline packages/office-ui-fabric-react/src/components/Button/Button.Props.ts]: Apparently we have a button that has both `onClick` and `menuProps`. This is why this PR introduces the `split` prop.
I'm currently creating a new `warn...` to make sure that if `split` is used, `onClick` and `menuProps` also are used.

Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/Button/Button.Props.ts]: How does a button with click and menuProps even render? Does the onClick just replace the "open/close" of the menu? Is there some other way to support this edge case?

Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/Button/Button.Props.ts]: i.e. at least make the 99% usecase the default and add some "suppressSplitButton" prop if you really do need to do that.

Comment (Boris-Em) [inline packages/office-ui-fabric-react/src/components/Button/Button.Props.ts]: I think that we want to avoid breaking people that were already using `onClick` and `menuProps` on a button. That's why we introduce this new prop.

Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/Button/Button.Props.ts]: works for me, though I'm still curious what usecase there is for this and if there is a better way to support it in the future.

File: packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx [hunk 4]
```diff
@@ -52,6 +53,12 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
   constructor(props: IBaseButtonProps, rootClassName: string) {
     super(props);
 
+    this._warnConditionallyRequiredProps(
+      ['menuProps', 'onClick'],
+      'split',
+      this.props.split === true
```
Comment (jair-cazarin) [inline packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx]: can this just be this.props.split?

Comment (Boris-Em) [inline packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx]: For sure.

File: packages/office-ui-fabric-react/src/components/Button/Button.Props.ts (hunk 3 above)
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Button/Button.Props.ts]: @micahgodbolt 

```tsx
onClick={ () => doTelemetry(); }
menuProps={ ... }
```
